pkg/providers/vsphere/defaults.go

```diff
@@ -101,12 +101,27 @@ func (d *Defaulter) setupDefaultTemplate(ctx context.Context, spec *Spec, machin
 	return nil
 }
 
-func (d *Defaulter) setCloneModeDefaults(ctx context.Context, machineConfig *anywherev1.VSphereMachineConfig, datacenter string) error {
+func max(a, b int) int {
+	if a > b {
+		return a
+	}
+	return b
+}
+
+//nolint:gocyclo // skip cyclomatic complexity check for this function
```

Comment:
This needs to be skipped? What is the error that lint is complaining about and how many is it?

Reply:
The [additional if condition](https://github.com/aws/eks-anywhere/pull/4798/files#diff-9d67b5f25923ee894de7e47cdf75a45f3224cde45ac7d518a97dfc8704bc8254R120-R123) bumps up the cyclomatic complexity to 11. I think 10 is the allowed value.

Reply:
Instead of bypassing the lint can we separate some part to a different function?

Reply:
yeah I split some of the logic in separate functions so it's good now